### ISSUE TITLE
TypeAheadField onTap Functionality

### ISSUE THREAD

**Issue by kjmj** [AbdulRahmanAlHamali/flutter_typeahead#144]
Just wanted to start off by saying that this is by far the best autocomplete widget for flutter. Very customizable, yet simple and easy to use. 

I was wondering if there is a way to access onTap for the `TypeAheadField`. We want to run some extra code when the user taps on the `TypeAheadField` widget, and can't figure out how to do this. We have tried using a focus node, but that seems to prevent any predictions (see #111).

Does anyone know of a way to run code when the user simply taps on the `TypeAheadField`? I appreciate any help and suggestions.

**Comment by Temperantia** [comment]
We agree this is by far the best Textfield wrapper available.

I ran into the same issue as you, and realised TypeAheadField was not providing an onTap binding for its internal TextField.

However, I have been able to perform a workaround thanks to a focus node I binded to the TypeAheadField. After that, I added an addListener to this node in my initState widget function and checked inside the callback if this node hasFocus or not.

It is a bit tricky but works well to detect an actual tap event, still I wish they would add onTap.
